text/0000-signed-transaction-payload/0000-signed-transaction-payload.md

```diff
@@ -0,0 +1,825 @@
+
+The following image depicts the flow of funds using UTXO:
+
+![UTXO flow](https://i.imgur.com/h3uxf6N.png)
+
+The way UTXOs are referenced is further described in the <i>Structure</i> section of this RFC.
+
+## Structure
+
+### Serialized Layout
+
+A <i>Signed Transaction</i> payload is made up of two parts:
+1. The <i>Unsigned Transaction</i> part which contains the inputs, outputs and an optional embedded payload.
+2. The <i>Unlock Blocks</i> which unlock the <i>Unsigned Transaction</i>'s inputs. In case the unlock block contains a signature, it signs the entire <i>Unsigned Transaction</i> part.
+
+All values are serialized in little-endian encoding. The serialized form of the transaction is deterministic, meaning the same logical transaction always results in the same serialized byte sequence.
+
+Following table structure describes the entirety of a <i>Signed Transaction</i> payload's serialized form:
+* [Data Type Notation](https://github.com/GalRogozinski/protocol-rfcs/blob/message/text/0017-message/0017-message.md#data-types)
+* <details>
+    <summary>Subschema Notation</summary>
+    <table>
+        <tr>
+            <th>Name</th>
+            <th>Description</th>
+        </tr>
+        <tr>
+            <td><code>oneOf</code></td>
+            <td>One of the listed subschemas.</td>
+        </tr>
+        <tr>
+            <td><code>optOneOf</code></td>
+            <td>Optionally one of the listed subschemas.</td>
+        </tr>
+        <tr>
+            <td><code>anyOf</code></td>
+            <td>Any (one or more) of the listed subschemas.</td>
+        </tr>
+    </table>
+</details>
+
+<p></p>
+
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>Payload Type</td>
+        <td>varint</td>
+        <td>
+        Set to <strong>value 0</strong> to denote a <i>Signed Transaction</i> payload.
+        </td>
+    </tr>
+    <tr>
+        <td valign="top">Transaction <code>oneOf</code></td>
+        <td colspan="2">
+            <details open="true">
+                <summary>Unsigned Transaction</summary>
+                <blockquote>
+                Describes the essence data making up a transaction by defining its inputs and outputs and an optional payload.
+                </blockquote>
+                <table>
+                    <tr>
+                        <td><b>Name</b></td>
+                        <td><b>Type</b></td>
+                        <td><b>Description</b></td>
+                    </tr>
+                    <tr>
+                        <td>Transaction Type</td>
+                        <td>varint</td>
+                        <td>
+                        Set to <strong>value 0</strong> to denote an <i>Unsigned Transaction</i>.
+                        </td>
+                    </tr>
+                   <tr>
+                        <td>Inputs Count</td>
+                        <td>varint</td>
```

Comment:
why use varint if you set the rule to be < 127? why not uint8?

Reply:
I think, the type of a length/count is a slightly different situation than a regular field. They represent serialized arrays of any kind and here always consistently serializing them by a prepended varint makes sense.
On of the main arguments of fixed integer types is not valid in this context as the length of the entire struct will always be dynamic due to the array.